lib/system/dollars.nim

```diff
@@ -160,3 +165,35 @@ proc `$`*[T](x: openArray[T]): string =
   ## .. code-block:: Nim
   ##   $(@[23, 45].toOpenArray(0, 1)) == "[23, 45]"
   collectionToString(x, "[", ", ", "]")
+
+proc `$`*[T: ref | ptr](arg: T): string =
```

Comment:
Not covered by an accepted RFC and worse the compiler still lacks the warning about using the default dollar when a specialized dollar is avaiable in a different module making the ``$`` more error prone than it already is.

Reply:
here is your RFC: https://github.com/nim-lang/RFCs/issues/203